src/docker/image.rs

```diff
@@ -0,0 +1,78 @@
+/// Represents a docker image URI such as `public.ecr.aws/myregistry/myrepo:v0.1.0`. The registry is
+/// optional as it is when using `docker`. That is, it will be looked for locally first, then at
+/// `dockerhub.io` when the registry is absent.
+#[derive(Default, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub(crate) struct ImageUri {
+    /// e.g. public.ecr.aws/bottlerocket
+    registry: Option<String>,
+    /// e.g. my-repo
+    repo: String,
+    /// e.g. v0.31.0
+    tag: String,
+}
+
+impl ImageUri {
+    /// Create a new `ImageUri`.
+    pub(crate) fn new<S1, S2>(registry: Option<String>, repo: S1, tag: S2) -> Self
+    where
+        S1: Into<String>,
+        S2: Into<String>,
+    {
+        Self {
+            registry,
+            repo: repo.into(),
+            tag: tag.into(),
+        }
+    }
+
+    /// Returns the `ImageUri` for use with docker, e.g. `public.ecr.aws/myregistry/myrepo:v0.1.0`
+    pub(crate) fn uri(&self) -> String {
```

Comment:
They're simple, but it's probably worth writing unit tests for these string formatting functions.